starfish/pipeline/features/spots/detector/gaussian.py

```diff
@@ -125,36 +97,38 @@ def _find_spot_locations(self) -> pd.DataFrame:
         fitted_blobs[Features.SPOT_RADIUS] = converted_radius
 
         # convert the array to int so it can be used to index
-        fitted_blobs = fitted_blobs.astype(int)
-
-        for v, max_size in zip(['z', 'y', 'x'], self.blobs_image.shape):
-            fitted_blobs[f'{v}_min'] = np.clip(
-                fitted_blobs[v] - fitted_blobs[Features.SPOT_RADIUS], 0, None)
-            fitted_blobs[f'{v}_max'] = np.clip(
-                fitted_blobs[v] + fitted_blobs[Features.SPOT_RADIUS], None, max_size)
+        rounded_blobs: pd.DataFrame = fitted_blobs.astype(int)
 
-        fitted_blobs['intensity'] = self._measure_blob_intensity(
-            self.blobs_image, fitted_blobs, self.measurement_function)
-        fitted_blobs['spot_id'] = np.arange(fitted_blobs.shape[0])
+        rounded_blobs['intensity'] = measure_spot_intensity(
+            data_image, rounded_blobs, self.measurement_function)
+        rounded_blobs['spot_id'] = np.arange(rounded_blobs.shape[0])
 
-        return fitted_blobs
+        return SpotAttributes(rounded_blobs)
 
-    def find(self, image_stack: ImageStack) -> IntensityTable:
-        """find spots
+    def find(self, data_stack: ImageStack, blobs_image: Union[np.ndarray, xr.DataArray]) \
+            -> IntensityTable:
+        """find spots in an ImageStack
 
         Parameters
         ----------
-        image_stack : ImageStack
+        data_stack : ImageStack
             stack containing spots to find
+        blobs_image : Union[np.ndarray, xr.DataArray]
 
         Returns
         -------
         IntensityTable :
             3d tensor containing the intensity of spots across channels and imaging rounds
 
         """
-        spot_attributes = self._find_spot_locations()
-        intensity_table = self._measure_spot_intensities(image_stack, spot_attributes)
+
+        intensity_table = detect_spots(
+            data_stack=data_stack,
+            spot_finding_method=self.image_to_spots,
+            reference_image=blobs_image,
```

Comment:
The find method in this detector is hard-coded to accept a 'blobs' image. This does not leverage the flexiblity of the detect_spots method to accept an argument where reference_image=None and reference_from_max_projection=True. This is a very common use case, and I think we should allow for this easily in the GaussianSpotDetector, especially if we go with the convention that the blobs_image should be normalized to [0,1] a-priori.

Reply:
Good catch, the typing should be Optional[...] to allow for the None case. 

You'll notice later in the PR that the tests do verify that the functionality you desire there works, so we just need to update the type 👍